src/components/parameters.tsx

```diff
@@ -5,76 +5,92 @@
-
 import { useSelector } from 'react-redux';
 import {
-    Grid,
     Box,
     Button,
     Container,
     Dialog,
     DialogContent,
     DialogTitle,
+    Grid,
     Tab,
     Tabs,
     Typography,
+    TypographyTypeMap,
 } from '@mui/material';
-
+import { CSSObject, Theme } from '@emotion/react';
 import { updateConfigParameter } from '../utils/rest-api';
 import { useSnackMessage } from '@gridsuite/commons-ui';
+import { AppState, AppStateKey } from '../redux/reducer';
 
 const styles = {
-    title: (theme) => ({
+    title: (theme: Theme): CSSObject => ({
         padding: theme.spacing(2),
     }),
-    grid: (theme) => ({
+    grid: (theme: Theme): CSSObject => ({
         padding: theme.spacing(2),
     }),
     controlItem: {
         justifyContent: 'flex-end',
-    },
+    } as CSSObject,
     button: {
         marginBottom: '30px',
-    },
+    } as CSSObject,
 };
 
-export function useParameterState(paramName) {
+export function useParameterState<K extends AppStateKey>(
+    paramName: K
+): [AppState[K], (value: AppState[K]) => void] {
     const { snackError } = useSnackMessage();
-
-    const paramGlobalState = useSelector((state) => state[paramName]);
-
+    const paramGlobalState = useSelector((state: AppState) => state[paramName]);
     const [paramLocalState, setParamLocalState] = useState(paramGlobalState);
 
     useEffect(() => {
         setParamLocalState(paramGlobalState);
     }, [paramGlobalState]);
 
     const handleChangeParamLocalState = useCallback(
-        (value) => {
+        (value: AppState[K]) => {
             setParamLocalState(value);
-            updateConfigParameter(paramName, value).catch((error) => {
-                setParamLocalState(paramGlobalState);
-                snackError({
-                    messageTxt: error.message,
-                    headerId: 'paramsChangingError',
+            updateConfigParameter(paramName, value as string) //TODO how to check/cast?
+                .catch((error) => {
+                    setParamLocalState(paramGlobalState);
+                    snackError({
+                        messageTxt: error.message,
+                        headerId: 'paramsChangingError',
+                    });
                 });
-            });
         },
         [paramName, snackError, setParamLocalState, paramGlobalState]
     );
 
     return [paramLocalState, handleChangeParamLocalState];
 }
 
-const Parameters = ({ showParameters, hideParameters }) => {
+export type ParametersProps = PropsWithChildren<{
+    showParameters: boolean;
+    hideParameters: () => void;
+}>;
+
+const Parameters: FunctionComponent<ParametersProps> = (props) => {
     const [tabIndex, setTabIndex] = useState(0);
 
-    function TabPanel(props) {
+    function TabPanel(
+        props: PropsWithChildren<
+            TypographyTypeMap<{ index: number; value: number }>['props']
+        >
+    ): ReactElement {
         const { children, value, index, ...other } = props;
```

Comment:
```
export type ParametersProps = {
    showParameters: boolean;
    hideParameters: () => void;
};

type TabPanelProps = {
    index: number;
    value: number;
} & TypographyProps<'div'>;

const Parameters: FunctionComponent<ParametersProps> = ({
    showParameters,
    hideParameters,
}) => {
    const [tabIndex, setTabIndex] = useState(0);

    function TabPanel({
        index,
        value,
        children,
        ...tabPanelProps
    }: TabPanelProps): ReactElement {
```

What about something like that ?

Reply:
And also in Typography, if {...other} is set after the properties it will override if some are commons, I'm not sure it is the wished behavior.

Reply:
> And also in Typography, if {...other} is set after the properties it will override if some are commons, I'm not sure it is the wished behavior.

dunno which one must be kept

Reply:
> ```
> // ...
> ```
> 
> What about something like that ?

Can you check the last review an say me if it's OK like that?

Reply:
perfect :)